app/models/general_info.rb

```diff
@@ -44,6 +43,10 @@ class GeneralInfo
             numericality: { only_integer: true, greater_than: 0, less_than_or_equal_to: 20,
                             message: "should be a positive number (1-20)" }
   validates :ori, ori: true
+  validates :address, presence: true,
+                      uniqueness: { scope: [:incident_date_str,
+                                            :incident_time_str, :ori],
+                                    message: 'has an Incident reported with the same date and time' }
```

Comment:
I'm wondering what the best message to put here is, to make it as clear as possible to the end user what the problem is.

Maybe something like `'has an existing incident reported for the same date and time - are you sure you didn't fill out this incident report already?'`

Ideally, if there's already an incident filled out by a different user, it should probably tell the current user who that is (e.g. "There's already an incident report filled out for this incident by Bob") - how hard do you think that would be to implement?